### DISCUSSION TITLE
[Miscellaneous] How can a rollup use multiple Data Availability layers?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [ethereum-optimism/developers#415] [Miscellaneous] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

From the [docs](https://docs.optimism.io/stack/components#layers):

> The Data Availability Layer defines where the raw inputs to an OP Stack based chain are published. **An OP Stack chain can use one or more Data Availability module to source its input data**. Because an OP Stack chain is derived from the Data Availability Layer, the Data Availability module(s) used have a significant impact on the security model of a system.

How would this work in practice? Would the data posted to both DA layers, using one of them as a backup? Or would it be posted to just one layer at a time?

**Comment by hash1go** [comment, marked as answer]
My understanding is OP Stack cannot handle two DA layers at the same time. Data would be posted to just one layer at a time.

cc: @sbvegan  
Could you please confirm this?

**Reply by PaulRBerg** [reply]
Thank you for your answer.

If this is the case, I suggest rectifying the docs to specify that OP Stack is currently not capable of supporting more than one DA layer.

In any case -  the theoretical question remains. How could a rollup handle two or more DA layers?

**Reply by hash1go** [reply]
Regarding the need to rectify the documentation, I will confirm with the OP team.

Meanwhile, according to the definition of `Rollup`, it needs to use a blockchain with a single consensus as both the Settlement and DA layers. Therefore, there is no rollup that handles more than one DA layer, making it necessary to consider the case of `Optimium`.

It is theoretically possible to adopt a backup DA layer to ensure redundancy or switch DA layers for some reason. As long as the data required to reconstruct the chain exists on an external chain, it should meet `Optimium`'s requirements, making it an engineering problem.

As for specific changes to the OP Stack, I am not sure as it is beyond my understanding.

**Reply by sbvegan** [reply]
Hey @PaulRBerg, this page is pretty old and could use a refresh. Its older than the "created" date because we migrated docs framework. Feel free to open an issue in the docs repo and we can have our docs person take a look. At the momenet, the OP stack just supports one DA layer at a time. We're testing a beta feature for our [plasma mode](https://specs.optimism.io/experimental/plasma.html) if you're interested in that. You could hack the op stack and modify it to use two da layers, but thats up to a curious mind to figure out.

**Reply by PaulRBerg** [reply]
Thanks for confirming @sbvegan. Created an issue in the docs repo: https://github.com/ethereum-optimism/docs/issues/713

**Comment by Chomtana** [comment]
We can post hashes using the PlasmaDA standard and post data to as many DA systems as we like. Data is then fetched through a PlasmaDA proxy. (As an idea)

There is a team working on something similar to this, [Plasma DA Translation Hub](https://app.charmverse.io/op-grants/plasmada-translation-hub-1014033406866528)
